### PR TITLE
#54506 - refactored Price component, added tests

### PR DESCRIPTION
**Pull request by szafran89**

### CODE REVIEW

File: src/atoms/price/Price.html [hunk 1]
```diff
@@ -1,24 +1,28 @@
-<div class="price">
-  <!-- @slot Custom old price slots (Named slots) -->
+<div class="a-price">
+  <!-- @slot Custom old price (Named slots) -->
```
Comment (robwozniak) [inline src/atoms/price/Price.html]: imo we can transform it to scoped slot and pass needed data, for example like `:special-price`, `:regular-price`. Please let me know what do you think, we can replace all named slots in this component to scoped.

File: src/atoms/price/Price.stories.js [hunk 2]
```diff
@@ -2,77 +2,30 @@ import { storiesOf } from '@storybook/vue'
...
-  .add('Special Price', () => ({
-    ...defaultData,
-    template: `
-      <a-price
-        :special-price="special"
-        aria-label-special="Special price:"
-      />
-    `
+    components: { APrice },
+    template: `<a-price regular-price="$36,00" />`
```
Comment (robwozniak) [inline src/atoms/price/Price.stories.js]: Would be great to have some knobs (even if they are not super useful) to use in this story for some values :slightly_smiling_face:

Comment (szafran89) [reply src/atoms/price/Price.stories.js]: knobs added

File: src/atoms/price/Price.html (hunk 1 above)
Comment (szafran89) [reply src/atoms/price/Price.html]: as we discussed through the zoom

**Pull request merged**